client/state/timezones/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { timezonesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Collect and store timezones data
+ *
+ * @param  {Object} state - Current state
+ * @param  {Object} action - action payload
+ * @return {Object} updated state
+ */
+export const items = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, {
+		manual_utc_offsets,
+		timezones_by_continent,
+	} ) => ( { manual_utc_offsets, timezones_by_continent } ),
+}, timezonesSchema );
```

Comment:
Please consider the structure of this reducer. In fact we are combining two different data properties into one reducer and as a result we have a complicated and unspecific initial state. If instead we use the idiomatic Redux composition we end up with separate reducers which can compose and automatically create a valid initial state.

```js
const offsets = ( state = [], action ) =>
	TIMEZONES_RECEIVE === action.type
		? action.manual_utc_offsets
		: state;

const timezones = ( state = [], action ) =>
	TIMEZONES_RECEIVE === action.type
		? action.timezones_by_continent
		: state;

const items = combineReducers( {
	offsets,
	timezones,
} )
```

of course, if you read my earlier comments I have some suggestions for restructuring the data which would have implications here.

please also consider the difference between _replacing_ operations and _merging_ operations. when we _replace_ we eliminate history and wipe out any changes. might it be possible that for testing or development purposes we might want to inject a specific timezone? if we _merge_ the arrays instead of replace them we can easily do this from the console

```js
dispatch( {
	type: actionTypes.TIMEZONES_RECEIVE,
	manual_utc_offset: [ { value: 'UTC-7', label: 'Dinner Time' } ]
} )
```

and ideally this just pops into the list without breaking anything else. didn't even need to add all the continents in the action to make it work.